src/MICore/CommandFactories/MICommandFactory.cs

```diff
@@ -659,6 +659,23 @@ public virtual bool CanDetach()
 
         abstract public Task<List<ulong>> StartAddressesForLine(string file, uint line);
 
+        public virtual async Task<string[]> AutoComplete(string command, int threadId, uint frameLevel)
+        {
+            command = "-complete \"" + command + "\"";
```

Comment:
Looking at [the documentation](https://sourceware.org/gdb/onlinedocs/gdb/GDB_002fMI-Miscellaneous-Commands.html#GDB_002fMI-Miscellaneous-Commands) for the `-complete` command, and a quick example, it looks like this just tells us about GDB commands, correct?

If so, don't we need to restrict this to when the command buffer starts with `-exec`

My quick example:
```
-complete "f"
^done,completion="f",matches=["faas","file","find","finish","flash-erase","flushregs","focus","forward-search","frame","fs","ftrace","function"],max_completions_reached="0"
```

Reply:
I did handle the short form ` since I can't be bothered to type -exec all the time but made it optional since I'm in the microsoft/vscode-cpptools#106 camp.
Gdb could be tricked into expression completion via the p command I think but for gdb command mode should really be the default.